x-pack/filebeat/input/streaming/websocket.go

```diff
@@ -283,10 +299,12 @@ func (s *websocketStream) FollowStream(ctx context.Context) error {
 					heartBeatCancel()
 				}
 				if !s.cfg.Retry.BlanketRetries && !isRetryableError(err) {
+					s.status.UpdateStatus(status.Failed, "failed to read websocket data: "+err.Error())
 					s.log.Errorw("failed to read websocket data", "error", err)
 					return err
 				}
 				s.log.Debugw("websocket connection encountered an error, attempting to reconnect...", "error", err)
+				s.status.UpdateStatus(status.Degraded, "websocket connection encountered an error: "+err.Error())
```

Comment:
So one question here, should we not wait for a certain number of failures to occur before we mark the status as Degraded ? The current philosophy seems to take a heavy handed approach as in 1st error -> update status as "degraded" -> if it reconnects then update status to "running".

Reply:
I think I'm OK with the current approach. We can revisit this if it is something that leads to UI/log churn. If we have a certain number other than one (a very certain number :smile:), we need to decide what it is, or allow a user configuration.